reana_workflow_controller/rest.py

```diff
@@ -1485,6 +1491,134 @@ def get_workflow_parameters(workflow_id_or_name):  # noqa
+          type: string
+      responses:
+        200:
+          description: >-
+            Request succeeded. Info about a workflow, including the status is
+            returned.
+          schema:
+            type: object
+            properties:
+              reana_specification:
+                type: string
+              workspace_listing:
+                type: string
+          examples:
+            application/json:
+              {
+                "reana_specification":
+                ["- nevents: 100000\n+ nevents: 200000"],
+                "workspace_listing": {"Only in workspace a: code"}
+              }
+        400:
+          description: >-
+            Request failed. The incoming payload seems malformed.
+          examples:
+            application/json:
+              {
+                "message": "Malformed request."
+              }
+        403:
+          description: >-
+            Request failed. User is not allowed to access workflow.
+          examples:
+            application/json:
+              {
+                "message": "User 00000000-0000-0000-0000-000000000000
+                            is not allowed to access workflow
+                            256b25f4-4cfb-4684-b7a8-73872ef455a1"
+              }
+        404:
+          description: >-
+            Request failed. Either user or workflow does not exist.
+          examples:
+            application/json:
+              {
+                "message": "Workflow 256b25f4-4cfb-4684-b7a8-73872ef455a1 does
+                            not exist."
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
+    try:
+        user_uuid = request.args['user']
+        brief = request.args.get('brief', False)
+        brief = True if brief == 'true' else False
+        context_lines = request.args.get('context_lines', 5)
+
+        workflow_a_exists = False
+        workflow_a = _get_workflow_with_uuid_or_name(workflow_id_or_name_a,
+                                                     user_uuid)
+        workflow_a_exists = True
+        workflow_b = _get_workflow_with_uuid_or_name(workflow_id_or_name_b,
+                                                     user_uuid)
+        if not workflow_id_or_name_a or not workflow_id_or_name_b:
+            raise ValueError("Workflow id or name is not supplied")
+        specification_diff = get_specification_diff(
+            workflow_a, workflow_b)
+        workspace_diff = get_workspace_diff(
+            workflow_a, workflow_b, brief, context_lines)
+        response = {'reana_specification': json.dumps(specification_diff),
+                    'workspace_listing': json.dumps(workspace_diff)}
+        return jsonify(response)
+    except REANAWorkflowControllerError as e:
+        return jsonify({"message": str(e)}), 409
+    except WorkflowInexistentError as e:
+        wrong_workflow = workflow_id_or_name_b if workflow_a_exists \
```

Comment:
Why do we need this auxiliary variable? Wouldn't it be enough to check if `workflow_id_or_name_a` is `None`?

Reply:
No need but it makes for better readability